src/blockdb/blockdb.h

```diff
@@ -0,0 +1,124 @@
+// Copyright (c) 2018 The Bitcoin Unlimited developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#ifndef BLOCKDB_H
+#define BLOCKDB_H
+
+#include "uint256.h"
+#include "dbwrapper.h"
+#include "chain.h"
+#include "primitives/block.h"
+#include "undo.h"
+
+struct BlockDBValue
+{
+    int32_t blockVersion;
+    uint64_t blockHeight;
+    CBlock block;
+
+    BlockDBValue()
+    {
+        SetNull();
+    }
+
+    BlockDBValue(const CBlock &_block)
+    {
+        assert(_block.IsNull() == false);
+        this->block = _block;
+        this->blockVersion = this->block.nVersion;
+        this->blockHeight = this->block.GetHeight();
+    }
+
+    template <typename Stream>
+    void Serialize(Stream &s) const
+    {
+        s << VARINT(blockVersion);
+        s << VARINT(blockHeight);
+        s << block;
+    }
+
+    template <typename Stream>
+    void Unserialize(Stream &s)
+    {
+        s >> VARINT(blockVersion);
+        s >> VARINT(blockHeight);
+        s >> block;
+    }
+
+    void SetNull()
+    {
+        blockVersion = 0;
+        blockHeight = 0;
+        block.SetNull();
+    }
+};
+
+struct UndoDBValue
+{
+    uint256 hashChecksum;
+    uint256 hashBlock;
+    CBlockUndo blockundo;
+
+    UndoDBValue()
+    {
+        SetNull();
+    }
+
+    UndoDBValue(const uint256 &_hashChecksum, const uint256 &_hashBlock, const CBlockUndo &_blockundo)
+    {
+        this->hashChecksum = _hashChecksum;
+        this->hashBlock = _hashBlock;
+        this->blockundo = _blockundo;
+    }
+
+    template <typename Stream>
+    void Serialize(Stream &s) const
+    {
+        s << FLATDATA(hashChecksum);
+        s << FLATDATA(hashBlock);
+        s << blockundo;
+    }
+
+    template <typename Stream>
+    void Unserialize(Stream &s)
```

Comment:
nit: `Deserialize`

Reply:
using unserialize keeps it consistent with the rest of the code. see serialize.h for examples

Reply:
fair enough. It was just that it sounded wrong to me :smile: